src/pages/Home.vue

```diff
@@ -38,11 +45,69 @@ export default {
       default: "",
       type: String,
     },
+    program: {
+      /** The program the user falls under. Eg: HaryanaStudents, DelhiStudents */
+      default: "",
+      type: String,
+    },
+    authType: {
+      /** The authentication method used by the user */
+      default: "",
+      type: String,
+    },
+  },
+  data() {
+    return {
+      textObject: {
+        placeholderText: "",
+        displayText: "",
+        addButtonText: "",
+        invalidInputMessage: "",
+        invalidLoginMessage: "",
+        submitButtonText: "",
+      },
+      maxNumberOfInput: 0,
+      inputObject: {
+        basicInputValidationType: "",
+        maxLengthOfId: 0,
+        mode: null,
+        type: null,
+      },
```

Comment:
suggesting a different name for the variable + that of inner params to avoid redundancy.

```suggestion
      inputParams: {
        basicValidationType: "",
        maxLength: 0,
        mode: null,
        type: null,
      },
```